spec/feature_flagger/storage/redis_spec.rb

```diff
@@ -3,63 +3,103 @@
 RSpec.describe FeatureFlagger::Storage::Redis do
   let(:redis)   { FakeRedis::Redis.new }
   let(:storage) { described_class.new(redis) }
-  let(:key)   { 'foo' }
-  let(:value) { 'bar' }
-  let(:global_key) { 'released_features' }
+  let(:feature_key)   { 'account:email_marketing:whitelabel' }
+  let(:resource_id)   { '1' }
+  let(:resource_name) { 'account' }
+  let(:global_key)    { 'released_features' }
+  let(:resource_key) do
+    FeatureFlagger::Storage::RedisKeys.resource_key(
+      FeatureFlagger::Storage::Redis::RESOURCE_PREFIX,
+      resource_name,
+      resource_id,
+    )
+  end
 
   context do
     before do
       redis.flushdb
     end
 
-    describe '#has_value?' do
-      context 'value is stored for given key' do
-        before { redis.sadd(key, value) }
-        it { expect(storage).to have_value(key, value) }
+    describe '#has_values?' do
+      context 'resource_id is stored for given feature_key' do
+        before { redis.sadd(feature_key, resource_id) }
+        it { expect(storage).to have_value(feature_key, resource_id) }
+      end
+
+      context 'resource_id is not stored for given feature_key' do
+        it { expect(storage).not_to have_value(feature_key, resource_id) }
+      end
+    end
+
+    describe '#fetch_releases' do
+      context 'when there is no features under global structure' do
+        before do
+          storage.add(feature_key, resource_name, resource_id)
+        end
+
+        it 'returns related feature_keys' do
+          expect(storage.fetch_releases(resource_name, resource_id, global_key)).to match_array([feature_key])
+        end
       end
 
-      context 'value is not stored for given key' do
-        it { expect(storage).not_to have_value(key, value) }
+      context 'when there is features under global structure' do
+        before do
+          storage.add(feature_key, resource_name, resource_id)
+          storage.add_all(global_key, feature_key)
+        end
+
+        it 'returns related feature_keys' do
+          expect(storage.fetch_releases(resource_name, resource_id, global_key)).to match_array(feature_key)
+        end
       end
     end
 
     describe '#add' do
-      it 'adds the value to redis' do
-        storage.add(key, value)
-        expect(redis.sismember(key, value)).to be_truthy
+      it 'adds the resource_id to redis' do
+        storage.add(feature_key, resource_name, resource_id)
+
+        expect(redis.sismember(feature_key, resource_id)).to be_truthy
+        expect(redis.sismember(resource_key, feature_key)).to be_truthy
```

Comment:
Why use the redis to make the assertion? And why there are two assertions and only one context?